azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/toolkit/lib/auth/Account.java

```diff
@@ -5,66 +5,82 @@
-import com.azure.core.util.logging.ClientLogger;
+import com.azure.identity.implementation.MsalToken;
+import com.azure.identity.implementation.util.IdentityConstants;
 import com.azure.identity.implementation.util.ScopeUtil;
 import com.azure.resourcemanager.AzureResourceManager;
 import com.azure.resourcemanager.resources.models.Tenant;
 import com.microsoft.aad.adal4j.AuthenticationException;
+import com.microsoft.aad.msal4j.IAuthenticationResult;
 import com.microsoft.azure.credentials.AzureTokenCredentials;
+import com.microsoft.azure.toolkit.lib.Azure;
 import com.microsoft.azure.toolkit.lib.account.IAccount;
+import com.microsoft.azure.toolkit.lib.auth.core.refresktoken.RefreshTokenTenantCredential;
 import com.microsoft.azure.toolkit.lib.auth.exception.AzureToolkitAuthenticationException;
-import com.microsoft.azure.toolkit.lib.auth.exception.LoginFailureException;
 import com.microsoft.azure.toolkit.lib.auth.model.AccountEntity;
 import com.microsoft.azure.toolkit.lib.auth.model.AuthMethod;
+import com.microsoft.azure.toolkit.lib.auth.util.AzureEnvironmentUtils;
 import com.microsoft.azure.toolkit.lib.common.model.Subscription;
 import com.microsoft.azure.toolkit.lib.common.utils.TextUtils;
 import com.microsoft.azure.toolkit.lib.common.utils.Utils;
 import lombok.Getter;
 import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.lang3.ObjectUtils;
 import org.apache.commons.lang3.StringUtils;
 import org.apache.commons.lang3.exception.ExceptionUtils;
+import org.apache.commons.lang3.reflect.FieldUtils;
+import reactor.core.publisher.Flux;
 import reactor.core.publisher.Mono;
+import reactor.core.scheduler.Schedulers;
+import reactor.util.function.Tuple2;
 
+import java.time.Duration;
 import java.util.ArrayList;
-import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.stream.Collectors;
 
 public abstract class Account implements IAccount {
-    private final ClientLogger logger = new ClientLogger(Account.class);
     @Getter
     protected AccountEntity entity;
 
     public Account() {
-        buildAccountEntity();
+        this.entity = new AccountEntity();
+        entity.setMethod(getMethod());
     }
 
     public abstract AuthMethod getMethod();
 
-    protected abstract Mono<Boolean> checkAvailableInner();
-
-    protected abstract void initializeCredentials() throws LoginFailureException;
+    protected abstract boolean checkAvailableInner();
 
-    boolean checkAvailable() {
-        checkAvailableInner().doOnSuccess(avail -> this.entity.setAvailable(avail)).onErrorContinue((e, i) -> this.entity.setLastError(e)).block();
-        return isAvailable();
-    }
+    protected abstract TokenCredential createTokenCredential();
 
     public AzureEnvironment getEnvironment() {
         return entity == null ? null : entity.getEnvironment();
     }
 
+    boolean checkAvailable() {
+        if (this.entity.isAvailable()) {
```

Comment:
Why we refresh the available status only if it is not available? Will it always be true once available?

Reply:
for avail: one check and then remain unchanged